lib/vrl/tests/tests/expressions/abort/abort_typedef.vrl

```diff
@@ -0,0 +1,31 @@
+# result: {
+#   "a": {"never": true},
+#   "b": {"bytes": true},
+#   "c": {"never": true},
+#   "d": {"object": {}},
+#   "e": {"null": true},
+#   "f": {"bytes": true, "integer": true},
+#   "g": {"never": true}
+#   }
+
+x = "string"
+
+.a = type_def({abort})
+.b = type_def({if false {abort} else {"string"}})
```

Comment:
What would the `type_def` look like if `false` was a boolean that is evaluated at runtime instead?

Reply:
It would be the same. The compiler doesn't do constant folding when calculating the type definition (yet). I'll add a true case also.